components/usage/pkg/contentservice/client.go

```diff
@@ -25,7 +28,33 @@ func New(url string) *Client {
 	return &Client{url: url}
 }
 
-func (c *Client) GetSignedUploadUrl(ctx context.Context) (string, error) {
+func (c *Client) UploadFile(ctx context.Context, filename string, body io.Reader) error {
+	url, err := c.getSignedUploadUrl(ctx, filename)
+	if err != nil {
+		return fmt.Errorf("failed to obtain signed upload URL: %w", err)
+	}
+
+	req, err := http.NewRequest(http.MethodPut, url, body)
+	if err != nil {
+		return fmt.Errorf("failed to construct http request: %w", err)
+	}
+
+	req.Header.Set("Content-Encoding", "gzip")
```

Comment:
Does this cause the content to be gzipped on the server? I would've expected you set this to indicate that the content you're uploading is `gzip` but I don't see us compressing in this implementation.

Reply:
See t[his comment](https://github.com/gitpod-io/gitpod/pull/11519#discussion_r928709112).